client/jetpack-cloud/sections/partner-portal/primary/terms-of-service-consent/index.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import React, { ReactElement, useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useMutation } from 'react-query';
+import { Button, Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { getQueryArg } from '@wordpress/url';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { APIPartner } from 'calypso/state/partner-portal/types';
+import wpcom from 'calypso/lib/wp';
+import {
+	getCurrentPartner,
+	hasFetchedPartner,
+	isFetchingPartner,
+} from 'calypso/state/partner-portal/partner/selectors';
+import { receivePartner } from 'calypso/state/partner-portal/partner/actions';
+import { errorNotice } from 'calypso/state/notices/actions';
+import {
+	ensurePartnerPortalReturnUrl,
+	formatApiPartner,
+} from 'calypso/jetpack-cloud/sections/partner-portal/utils';
+import QueryJetpackPartnerPortalPartner from 'calypso/components/data/query-jetpack-partner-portal-partner';
+import Main from 'calypso/components/main';
+import CardHeading from 'calypso/components/card-heading';
+import Spinner from 'calypso/components/spinner';
+
+function mutationConsent( consent: boolean ): Promise< APIPartner > {
+	return wpcom.req.post( {
+		method: 'PUT',
+		apiNamespace: 'wpcom/v2',
+		path: '/jetpack-licensing/partner',
+		body: { tos: consent },
+	} );
+}
+
+export default function TermsOfServiceConsent(): ReactElement | null {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const consent = useMutation( mutationConsent, {
+		onSuccess: ( partner ) => {
+			dispatch( receivePartner( formatApiPartner( partner ) ) );
+		},
+		onError: () => {
+			dispatch(
+				errorNotice( translate( 'We were unable to update your partner account details.' ) )
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'We were unable to retrieve your partner account details.' )` *ES Score: 8*